src/controllers/dynakube/activegate/internal/statefulset/statefulset.go

```diff
@@ -124,6 +125,37 @@ func buildTolerations(capability capability.Capability) []corev1.Toleration {
 	return tolerations
 }
 
+func buildTopologySpreadConstraints(capability capability.Capability) []corev1.TopologySpreadConstraint {
+	if len(capability.Properties().TopologySpreadConstraints) > 0 {
+		return capability.Properties().TopologySpreadConstraints
+	}
+
+	return defaultTopologyConstraints()
+}
+
+func defaultTopologyConstraints() []corev1.TopologySpreadConstraint {
+	labelSelector := &metav1.LabelSelector{
```

Comment:
you could use the labelSelector created in the `addLabels` step, so it should already be part of the spec of the statefulset

Reply:
then you just reuse the labelSelectors present in the statefulset